tests/HUBActionRegistryTests.m

```diff
@@ -57,8 +57,9 @@ - (void)testRegisteringFactoryAndCreatingAction
     HUBActionMock * const action = [[HUBActionMock alloc] initWithBlock:nil];
     HUBActionFactoryMock * const factory = [[HUBActionFactoryMock alloc] initWithActions:@{actionIdentifier.namePart: action}];
     [self.actionRegistry registerActionFactory:factory forNamespace:actionIdentifier.namespacePart];
-    
-    XCTAssertEqual([self.actionRegistry createCustomActionForIdentifier:actionIdentifier], action);
+
+    // These should be the same instance.
+    XCTAssertTrue([self.actionRegistry createCustomActionForIdentifier:actionIdentifier] == action);
```

Comment:
I guess it's because in a test, if I look at a line that says `XCTAssertEqual(objA, obj2)`, I immediately think "do we really want to check equality here, or is this a typo".

Admittedly `XCTAssertTrue(obj1 == obj2)` doesn't do a great deal to fix that ambiguity, which is why I like Hamcrest's approach of providing a "same instance" matcher for objects so that it's obvious by looking at it whether you want to check for equality or instance equivalence.